patches/tModLoader/Terraria/Item.TML.cs

```diff
@@ -17,11 +17,13 @@ public partial class Item : TagSerializable
 
 		private DamageClass _damageClass = DamageClass.Generic;
 		/// <summary>
-		/// The damage type of this Item. Assign to DamageClass.Melee/Ranged/Magic/Summon/Throwing for vanilla classes, or ModContent.GetInstance<T>() for custom damage types.
+		/// The damage type of this Item.
+		/// Assign to DamageClass.Melee/Ranged/Magic/Summon/Throwing for vanilla classes, or ModContent.GetInstance<T>() for custom damage types.
+		/// Set to DamageClass.Generic to not have any one damage class, but still have the weapon be boosted by universal bonuses.
 		/// </summary>
 		public DamageClass DamageType {
 			get => _damageClass;
-			set => _damageClass = value ?? throw new ArgumentException("DamageType cannot be null");
+			set => _damageClass = value;
```

Comment:
This isn't a fix, this is a rollback.

Reply:
The actual fix would be to not autoload the default classes with the default mod. Why are they done so..

Reply:
![image](https://user-images.githubusercontent.com/6795251/105628788-105f2e80-5e50-11eb-866c-1be514b9d736.png)